sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/jsonpatch/JsonPatchDocument.java

```diff
@@ -0,0 +1,210 @@
+     * previous value is replaced. If the {@code path} specifies an array index the value is inserted at the specified.
+     * <p>
+     * See <a href="https://tools.ietf.org/html/rfc6902#section-4.1">JSON Patch Add</a> for more information.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.core.experimental.jsonpatch.JsonPatchDocument.appendAdd#String-String}
+     *
+     * @param path The path to apply the addition.
+     * @param rawJsonValue The raw JSON value to add to the path.
+     * @return The updated JsonPatchDocument object.
+     * @throws NullPointerException If {@code path} or {@code rawJsonValue} is null.
+     */
+    public JsonPatchDocument appendAdd(String path, String rawJsonValue) {
+        operations.add(new JsonPatchOperation(JsonPatchOperationKind.ADD,
+            Objects.requireNonNull(path, "'path' cannot be null."), null,
+            Objects.requireNonNull(rawJsonValue, "'rawJsonValue' cannot be null.")));
+        return this;
+    }
+
+    /**
+     * Appends a "replace" operation to this JSON Patch document.
+     * <p>
+     * See <a href="https://tools.ietf.org/html/rfc6902#section-4.3">JSON Patch replace</a> for more information.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.core.experimental.jsonpatch.JsonPatchDocument.appendReplace#String-String}
+     *
+     * @param path The path to replace.
+     * @param rawJsonValue The raw JSON value to use as the replacement.
+     * @return The updated JsonPatchDocument object.
+     * @throws NullPointerException If {@code path} or {@code rawJsonValue} is null.
+     */
+    public JsonPatchDocument appendReplace(String path, String rawJsonValue) {
+        operations.add(new JsonPatchOperation(JsonPatchOperationKind.REPLACE,
+            Objects.requireNonNull(path, "'path' cannot be null."), null,
+            Objects.requireNonNull(rawJsonValue, "'rawJsonValue' cannot be null.")));
+        return this;
+    }
+
+    /**
+     * Appends a "copy" operation to this JSON Patch document.
+     * <p>
+     * See <a href="https://tools.ietf.org/html/rfc6902#section-4.5>JSON Patch copy</a> for more information.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.core.experimental.jsonpatch.JsonPatchDocument.appendCopy#String-String}
+     *
+     * @param from The path to copy from.
+     * @param path The path to copy to.
+     * @return The updated JsonPatchDocument object.
+     * @throws NullPointerException If {@code from} or {@code path} is null.
+     */
+    public JsonPatchDocument appendCopy(String from, String path) {
+        operations.add(new JsonPatchOperation(JsonPatchOperationKind.COPY,
+            Objects.requireNonNull(path, "'path' cannot be null."),
+            Objects.requireNonNull(from, "'from' cannot be null."), null));
```

Comment:
nit: too much happening in this 1 statement - add, new & null checks. Breaking it down probably makes it easier to read.

Reply:
Separated out all `Objects.requireNonNull` to their own lines.